### PR TITLE
Port ZStream.fromInputStream (the one in platform.scala) to StreamEffect #1460

### PR DESCRIPTION
**Pull request by jsilva**
Pull request for #1460

### CODE REVIEW

File: streams/shared/src/main/scala/zio/stream/StreamEffect.scala [hunk 1]
```diff
@@ -320,6 +322,25 @@ private[stream] object StreamEffect extends Serializable {
       }
     }
 
+  final def fromInputStream(is: InputStream): StreamEffect[Any, Nothing, Byte] =
+    StreamEffect[Any, Nothing, Byte] {
+      Managed.effectTotal {
+        def pull(): Byte = {
+          val a = is.read()
```
Comment (regiskuckaertz) [inline streams/shared/src/main/scala/zio/stream/StreamEffect.scala]: this can throw an `IOException`, why not `StreamEffect[Any, IOException, Byte]` then?

Comment (iravid) [reply streams/shared/src/main/scala/zio/stream/StreamEffect.scala]: Yes, good point. We need to try/catch the `IOException` and re-throw it using in `StreamEffect.fail`.

File: streams/shared/src/main/scala/zio/stream/StreamEffect.scala [hunk 2]
```diff
@@ -322,19 +331,22 @@ private[stream] object StreamEffect extends Serializable {
...
+    StreamEffect[Any, IOException, Chunk[Byte]] {
       Managed.effectTotal {
-        def pull(): Byte = {
-          val a = is.read()
-          if (a < 0) {
-            end
-          } else {
-            a.toByte
-          }
+        val buf = Array.ofDim[Byte](chunkSize)
```
Comment (iravid) [inline streams/shared/src/main/scala/zio/stream/StreamEffect.scala]: The `buf` has to be re-created every time `pull` runs, or we'll be mutating chunks that have already been emitted.

File: streams/shared/src/main/scala/zio/stream/StreamEffect.scala [hunk 3]
```diff
@@ -322,19 +331,22 @@ private[stream] object StreamEffect extends Serializable {
...
+        def pull(): Chunk[Byte] = {
+          val bytesRead = is.read(buf)
```
Comment (iravid) [inline streams/shared/src/main/scala/zio/stream/StreamEffect.scala]: this needs to be wrapped in try/catch and should catch `IOException` and re-throw it  using `StreamEffect.fail`

File: streams/shared/src/main/scala/zio/stream/StreamEffect.scala [hunk 4]
```diff
@@ -322,19 +331,22 @@ private[stream] object StreamEffect extends Serializable {
       }
     }
 
-  final def fromInputStream(is: InputStream): StreamEffect[Any, Nothing, Byte] =
-    StreamEffect[Any, Nothing, Byte] {
+  final def fromInputStream(
```
Comment (iravid) [inline streams/shared/src/main/scala/zio/stream/StreamEffect.scala]: I think this needs to go into a JVM-specific file?

File: streams/shared/src/main/scala/zio/stream/StreamEffect.scala [hunk 5]
```diff
@@ -320,6 +322,25 @@ private[stream] object StreamEffect extends Serializable {
       }
     }
 
+  final def fromInputStream(is: InputStream): StreamEffect[Any, Nothing, Byte] =
+    StreamEffect[Any, Nothing, Byte] {
+      Managed.effectTotal {
```
Comment (regiskuckaertz) [inline streams/shared/src/main/scala/zio/stream/StreamEffect.scala]: This is not a total effect, I can see `Managed.makeEffect` which is probably more appropriate here.

File: streams/shared/src/main/scala/zio/stream/StreamEffect.scala (hunk 4 above)
Comment (jsilva) [reply streams/shared/src/main/scala/zio/stream/StreamEffect.scala]: I am not sure, however scala.js has https://github.com/scala-js/scala-js/blob/master/javalib/src/main/scala/java/io/InputStream.scala

Comment (iravid) [reply streams/shared/src/main/scala/zio/stream/StreamEffect.scala]: Oh well that simplifies everything. We can keep it here then :-)

File: streams/shared/src/main/scala/zio/stream/StreamEffect.scala (hunk 5 above)
Comment (iravid) [reply streams/shared/src/main/scala/zio/stream/StreamEffect.scala]: Well the `Managed` only returns a thunk that, so in that regard it *should* be total

File: streams/shared/src/main/scala/zio/stream/StreamEffect.scala (hunk 2 above)
Comment (jsilva) [reply streams/shared/src/main/scala/zio/stream/StreamEffect.scala]: Updated this

File: streams/shared/src/main/scala/zio/stream/StreamEffect.scala (hunk 3 above)
Comment (jsilva) [reply streams/shared/src/main/scala/zio/stream/StreamEffect.scala]: updated, thanks for help

File: streams/shared/src/main/scala/zio/stream/StreamEffect.scala (hunk 5 above)
Comment (vasilmkd) [reply streams/shared/src/main/scala/zio/stream/StreamEffect.scala]: This is how we treated all `StreamEffect` code so far.

**Pull request merged**